core/internal/filetransfer/file_transfers.go

```diff
@@ -28,17 +30,11 @@ func NewFileTransfers(
 	logger *observability.CoreLogger,
 	fileTransferStats FileTransferStats,
 ) *FileTransfers {
-	filetransfers := &FileTransfers{}
-
 	defaultFileTransfer := NewDefaultFileTransfer(client, logger, fileTransferStats)
-	filetransfers.Default = defaultFileTransfer
+	gcsFileTransfer := NewGCSFileTransfer(nil, logger, fileTransferStats)
```

Comment:
it is usually common to leave a comment if you pass a nil argument:
gcsFileTransfer := NewGCSFileTransfer(nil /* ...*/, logger, fileTransferStats)

Reply:
will follow up with another PR to add this!